Ocr crmops p5

crm-ops opened the pull request:
@fpatel,

Please have a look at these commits. 

Kind Regards

Sébastien

Commit:
P5 - svige - 28/12/2020

- added sl4j logging support
- fixed junit exception due to missing package

Commit:
P5 - svige - 28/12/2020

- fixed unit test by implmeenting jodatime
- fixed existing integration test by adding junit maven dependency
- added a funcional testing plan to be implemented

Commit:
P5 - svige - 04/01/2020

 - ammended processExitingVehicleTest()
 - added processIncomingVehicleTest()

Am I on the the correct path ? I have been struggling a bit on this.

Commit:
Added processIncomingVehiculeExceptionTest
 - test passes with a MockitoException I cannot solve

Commit:
fixed pomxml with jupiter dependency
jacoco report are generated again

Commit:
100% Coverage on parkingService
All unit tests passing - still some exception in the logs
Not sure methodology / good practices respected

Commit:
100% Coverage on parkingService
All unit tests passing - still some exception in the logs
Not sure methodology / good practices respected
190CET - Fixed process incoming vehicule exception Test

Commit:
100% Coverage on parkingService
All unit tests passing - no more exception in the logs
Exceptions in test code removed
Exception in main code optimised following call stack principle
Use the following partern :
==> unchecked exception - handled locally in the class
==> checked exception > thrown in the child class - handled in the stack to allow app recovery (restarting menu / last input from user)
==> reviewed and cleaned unit tests

Commit:
11 jan 2020
Added two scenarii with 30 mins free & 5% discount
TODO finalise existing basic integration test
TODO add all functional integration tests

Commit:
12-Jan-2021
P5_Final_commit_pre_review
All unit tests completed and passing including new features
Coverage 100% on ParkingService
Created custom exceptions for some functional cases
Integration Test complete and passing

TODO - Review Code (Faizal)
TODO - Presentation for Assessment (Seb)

fpatelm commented on src/test/java/com/parkit/parkingsystem/integration/ParkingDataBaseIT.java:
```diff
@@ -33,33 +43,211 @@ private static void setUp() throws Exception{
+            //covered in unit test
+    // test 5% discount for returning client
+            //covered in unit test
+
```
You can remove these comments or place them relatively to each test

fpatelm commented on src/test/java/com/parkit/parkingsystem/integration/ParkingDataBaseIT.java:
```diff
@@ -33,33 +43,211 @@ private static void setUp() throws Exception{
+        when(inputReaderUtil.readVehicleRegistrationNumber()).thenReturn("ABCDEF");
         ParkingService parkingService = new ParkingService(inputReaderUtil, parkingSpotDAO, ticketDAO);
         parkingService.processIncomingVehicle();
         //TODO: check that a ticket is actualy saved in DB and Parking table is updated with availability
```
is this TODO done?

fpatelm commented on src/test/java/com/parkit/parkingsystem/integration/ParkingDataBaseIT.java:
```diff
@@ -33,33 +43,211 @@ private static void setUp() throws Exception{
-        when(inputReaderUtil.readSelection()).thenReturn(1);
-        when(inputReaderUtil.readVehicleRegistrationNumber()).thenReturn("ABCDEF");
-        dataBasePrepareService.clearDataBaseEntries();
+      //  when(inputReaderUtil.readSelection()).thenReturn(1);
```
remove not used code

fpatelm commented on src/main/java/com/parkit/parkingsystem/util/InputReaderUtil.java:
```diff
@@ -1,38 +1,54 @@
-import org.apache.logging.log4j.Logger;
+// svige replace Log4j support by sl4j
+//import org.apache.logging.log4j.LogManager;
+//import org.apache.logging.log4j.Logger;
```
this can be removed

fpatelm reviewed (changes requested):
Minor changes, basically removing commented code and comments.
Good Job!

Commit:
16-Jan-2021
P5_Final_commit_post_review

@fpatelm - all discussions resolved
run test suites unitary & integration for non-regression - all passing green

crm-ops commented:
@fpatelm 
Thank you for the review comments - all redundant comments removed - test suites run all passing - please review and validate project.
